### ISSUE TITLE
[2513.2.0 beta] udev rule for md-raid: invalid value

### ISSUE THREAD

**Issue by vbatts** [flatcar/Flatcar#164]
**Description**

looking at errors in journal on a fresh boot of 2513.2.0 beta

**Impact**

from `journalctl`
```
Aug 12 16:11:58 localhost systemd[1]: Starting udev Kernel Device Manager...
Aug 12 16:11:58 localhost systemd-udevd[659]: /usr/lib64/udev/rules.d/11-dm-lvm.rules:40 Invalid value for OPTIONS key, ignoring: 'event_timeout=180'
Aug 12 16:11:58 localhost systemd-udevd[659]: /usr/lib64/udev/rules.d/11-dm-lvm.rules:40 The line takes no effect, ignoring.
Aug 12 16:11:58 localhost systemd-udevd[659]: /usr/lib64/udev/rules.d/64-md-raid-assembly.rules:30 Invalid value "/sbin/mdadm --incremental --export $devnode --offroot ${DEVLINKS}" for IMPORT (char 55: invalid substitution type), ignoring, but please fix it.
```

**Environment and steps to reproduce**

1. **Set-up**: flatcar_production_qemu_image.img.bz2
2. **Task**: review `journalctl`

**Expected behavior**
valid rules loaded

**Comment by krnowak** [comment]
About `11-dm-lvm.rules`: Support for `OPTIONS+="event_timeout=180"` was removed in https://github.com/flatcar-linux/systemd/commit/dd5eddd28a74a49607a8fffcaf960040dba98479, so this could be fixed by either patching the rules file or updating the `sys-fs/lvm2` in coreos-overlay. The fix in upstream is here: https://sourceware.org/git/?p=lvm2.git;a=commit;h=125f27ac37bc9b93cc96f64052b9681b3d479ee1.

About `64-md-raid-assembly.rules`: It's a bug in mdadm - it uses `${DEVLINKS}` instead of `$env{DEVLINKS}`. The fix in upstream is here: https://git.kernel.org/pub/scm/utils/mdadm/mdadm.git/commit/?id=ceb5f8ef92c97d1f44c75a3b74f64aa12a3303ef. Since the package is in portage-stable, I'm tempted to fix it by updating to 4.1 (currently it's 3.4).

**Comment by vbatts** [comment]
mdadm: take the 4.1 update.

dm-lvm:  while I don't think updating lvm2 is a bad idea, that may have more issues that just dropping that value like upstream did.

**Comment by dongsupark** [comment]
@krnowak Great findings!

> About 64-md-raid-assembly.rules: It's a bug in mdadm - it uses ${DEVLINKS} instead of $env{DEVLINKS}. The fix in upstream is here: https://git.kernel.org/pub/scm/utils/mdadm/mdadm.git/commit/?id=ceb5f8ef92c97d1f44c75a3b74f64aa12a3303ef. Since the package is in portage-stable, I'm tempted to fix it by updating to 4.1 (currently it's 3.4).

You are right.
However, while Alpha has mdadm 3.4 in portage-stable, Edge has mdadm [4.1](https://github.com/flatcar-linux/coreos-overlay/tree/7e71177a86568f1a0bb7a8911b7459322ee755c8/sys-fs/mdadm) in coreos-overlay with Flatcar-specific patches to support the default layout. Reason for adding it was to avoid RAID0 disk layout issues introduced in Kernel 3.14. (see the [mdadm commit](https://git.kernel.org/pub/scm/utils/mdadm/mdadm.git/commit/?id=329dfc28debb58ffe7bd1967cea00fc583139aca) for details, also our [PR](https://github.com/flatcar-linux/coreos-overlay/pull/146))

8 months ago, I expected the RAID0 layout patches would be "soon" included in another release like 4.2. However, the latest mdadm release is still 4.1 until now. :facepalm: So it is not possible to simply update mdadm to the latest version in portage-stable, and at the same time to keep the third-party patches.

So I would like to suggest updating mdadm 4.1 in coreos-overlay for the main branch and include the default layout patches, just like Edge.
That would help us maintain Edge, as Alpha could be aligned with Edge.
